deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/util/ValidationUtils.java

```diff
@@ -0,0 +1,332 @@
+package org.deeplearning4j.util;
+
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+
+public class ValidationUtils {
+
+    /**
+     * Checks that the values is >= 0.
+     *
+     * @param data An int
+     * @param paramName The param name, for error reporting
+     */
+    public static void validateNonNegative(int data, String paramName){
+        Preconditions.checkArgument(data >= 0,
+                "Values for " +
+                        paramName + " must be >= 0, got: " +
+                        data);
```

Comment:
This comment applies to all of your uses of Preconditions case: don't create strings like this.
The correct way of using preconditions class doesn't (in most cases) result in any object creation unless the precondition fails:
```
Preconditions.checkArgument(data >= 0, "Values for %s must be >= 0, got %s", paramName, data);
```